src/ray/core_worker/task_manager.h

```diff
@@ -44,6 +44,7 @@ class TaskManager : public TaskFinisherInterface {
   /// \param[in] task_id ID of the task to query.
   /// \return Whether the task is pending.
   bool IsTaskPending(const TaskID &task_id) const {
+    absl::MutexLock lock(&mu_);
```

Comment:
This seems to have evaded the thread annotation checks, perhaps since the method is declared in the header.

Reply:
Good catch. How about we move it to .cc to make sure it can be covered by auto-check?

Reply:
Ah yes I came across this too - method bodies declared in the header seem to not be covered at all. We should probably just avoid writing any code in headers to avoid this.